### ISSUE TITLE
Select all across pages pattern for tables and lists 

### ISSUE THREAD

**Issue by rachael-phillips** [patternfly/patternfly-next#1878]
There is a need for a pattern for selecting items across pages in a list. [Aka, when there are 2+ pages of data, it is unclear if select all applies to just the visible page, or the entire list. Gmail has a pretty good pattern for this]

This is an enhancement to the "select all" checkbox that is already available in the data table & toolbar.

Cloud doc: https://sketch.cloud/s/JaL1e
Design file: https://drive.google.com/file/d/1Bx5gjpMj4DghnavcwF0j5svcaLFOKVM8/view?usp=sharing

cc @katierik

**Comment by katierik** [comment]
I have been working with this google doc to track research & the design proposal:
https://docs.google.com/document/d/1o6Uippbb2gCrMOYzzJ1kQmXOLRY-SucofanDpKLxcYQ/edit?usp=sharing

**Comment by rachael-phillips** [comment]
@mcarrano and @LHinson making you aware of this. @mcarrano have we seen a need for this before or across applications?

**Comment by mcarrano** [comment]
Yes. And this is one of the use cases that is being addressed here: https://github.com/patternfly/patternfly-next/issues/1089 as described in the doc by @katierik above. I suggest that this be closed as a duplicate unless you think this is different from what you are working on @katierik

**Comment by rachael-phillips** [comment]
@katierik would you like me to close this one or keep it open?

**Comment by katierik** [comment]
Open as per the discussion at the last pattern meeting about keeping the #1089 and this separate.

Here are designs for *selection only*
https://marvelapp.com/i50gj84

**Comment by rachael-phillips** [comment]
Thank you!

**Comment by mcarrano** [comment]
I've tagged this as a demo. Once we have the bulk selector component, it would be great to have a demo that shows how this works in context.

**Comment by mcarrano** [comment]
Updated the design link to reflect what I believe is the latest design.
